test/reservation_test.rb

```diff
@@ -0,0 +1,103 @@
+require_relative 'test_helper'
+
+describe "Reservation class" do
+  before do
+    @date = Date.parse("2020-02-01")
```

Comment:
Instead of parsing `Date`s from strings here you should just directly instantiate them:

```suggestion
    @date = Date.new(2020, 2, 1)
```